### PR TITLE
Bump 4w compat to v1.7.0 release Sept 05, 2024

### PR DESCRIPTION
**Pull request by abhigunj**
fixes https://github.com/openxla/stablehlo/issues/2578

It is safe to bump 4w compat to `1.6.0`.  It is more that 4 weeks since it's release date (Aug 16, 2024).

### CODE REVIEW

**Comment by apivovarov** [comment]
Can you bump it to 1.7.0? it was released 29 days ago which is more that 4 weeks.

File: stablehlo/dialect/Version.cpp [hunk 1]
```diff
@@ -80,7 +80,7 @@ Version Version::fromCompatibilityRequirement(
     case CompatibilityRequirement::NONE:
       return Version::getCurrentVersion();
     case CompatibilityRequirement::WEEK_4:
-      return Version(1, 5, 0);  // v1.5.0 - Aug 1, 2024
+      return Version(1, 6, 0);  // v1.6.0 - Aug 16, 2024
```
Comment (GleasonK) [inline stablehlo/dialect/Version.cpp:83]: Above this switch statement, can you add a comment:

```
// The time frames used are from the date that the release was tagged on, not merged.
// The tag date is when the version has been verified and exported to XLA.
// See: https://github.com/openxla/stablehlo/tags
```

Comment (GleasonK) [reply stablehlo/dialect/Version.cpp:83]: Also bump to 1.7.0 as mentioned, that will allow new fp types to be used from JAX export, that tag date is now 4w ago.

**Comment by apivovarov** [comment]
Thank you for updating it so quickly!

**Pull request merged**